x-pack/plugins/ml/public/application/jobs/new_job/job_from_dashboard/quick_create_job_base.ts

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { mergeWith, uniqBy, isEqual } from 'lodash';
+import type { IUiSettingsClient } from '@kbn/core/public';
+import type { TimefilterContract } from '@kbn/data-plugin/public';
+import type { SharePluginStart } from '@kbn/share-plugin/public';
+import { firstValueFrom } from 'rxjs';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { DashboardAppLocatorParams } from '@kbn/dashboard-plugin/public';
+import type { Filter, Query, DataViewBase } from '@kbn/es-query';
+import { FilterStateStore } from '@kbn/es-query';
+import type { Embeddable } from '@kbn/lens-plugin/public';
+import type { MapEmbeddable } from '@kbn/maps-plugin/public';
+import type { MlApiServices } from '../../../services/ml_api_service';
+import { getFiltersForDSLQuery } from '../../../../../common/util/job_utils';
+import type { ErrorType } from '../../../../../common/util/errors';
+import { CREATED_BY_LABEL } from '../../../../../common/constants/new_job';
+import { createQueries } from '../utils/new_job_utils';
+import { createDatafeedId } from '../../../../../common/util/job_utils';
+import { Job, Datafeed } from '../../../../../common/types/anomaly_detection_jobs';
+
+export function isLensEmbeddable(arg: any): arg is Embeddable {
+  return arg.hasOwnProperty('type') && arg.type === 'lens';
+}
+
+export function isMapEmbeddable(arg: any): arg is MapEmbeddable {
+  return arg.hasOwnProperty('type') && arg.type === 'map';
+}
+
+export type Dashboard = Embeddable['parent'];
+
+interface CreationState {
+  success: boolean;
+  error?: ErrorType;
+}
+
+export interface CreateState {
+  jobCreated: CreationState;
+  datafeedCreated: CreationState;
+  jobOpened: CreationState;
+  datafeedStarted: CreationState;
+}
+
+export class QuickJobCreatorBase {
+  constructor(
+    protected readonly kibanaConfig: IUiSettingsClient,
+    protected readonly timeFilter: TimefilterContract,
+    protected readonly share: SharePluginStart,
+    protected readonly mlApiServices: MlApiServices
+  ) {}
+
+  protected async putJobAndDataFeed({
+    jobId,
+    datafeedConfig,
+    jobConfig,
+    createdByLabel,
+    dashboard,
+    start,
+    end,
+    startJob,
+    runInRealTime,
+  }: {
+    jobId: string;
+    datafeedConfig: Datafeed;
+    jobConfig: Job;
+    createdByLabel: CREATED_BY_LABEL;
+    dashboard: any;
```

Comment:
```suggestion
    dashboard: Dashboard;
```

Reply:
Updated in 8da1fadcaca00341a6a40c9abda84d799ceddd95